packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -213,14 +264,20 @@ export default function({ types: t }) {
             );
           }
 
-          const [argument, callExpression] = createObjectSpread(
-            file,
-            path.node.left.properties,
-            ref,
-          );
+          const [
+            impureComputedPropertyDeclarators,
+            argument,
+            callExpression,
+          ] = createObjectSpread(leftPath, file, ref);
+
+          if (impureComputedPropertyDeclarators.length > 0) {
+            nodes.push(
+              t.variableDeclaration("var", impureComputedPropertyDeclarators),
+            );
+          }
 
           const nodeWithoutSpread = t.clone(path.node);
-          nodeWithoutSpread.right = ref;
+          nodeWithoutSpread.right = ref; //todo: shouldn't the if statement always execute for this to work
```

Comment:
Yes. The above if statement should always run. http://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&targets=&browsers=&builtIns=false&debug=false&code_lz=C4UwzsAUkN4IYBoAEA6NcC-SC8SBGAlMgIwFA